gazelle/manifest/defs.bzl

```diff
@@ -7,21 +7,52 @@ load("@io_bazel_rules_go//go:def.bzl", "go_binary")
 def gazelle_python_manifest(
         name,
         requirements,
-        pip_deps_repository_name,
         modules_mapping,
+        pip_repository_name = "",
+        pip_repository_incremental = False,
+        pip_deps_repository_name = "",
         manifest = ":gazelle_python.yaml"):
     """A macro for defining the updating and testing targets for the Gazelle manifest file.
 
     Args:
         name: the name used as a base for the targets.
         requirements: the target for the requirements.txt file.
-        pip_deps_repository_name: the name of the pip_install repository target.
+        pip_repository_name: the name of the pip_install or pip_repository target.
+        pip_repository_incremental: the incremental property of pip_repository.
+        pip_deps_repository_name: deprecated - the old pip_install target name.
         modules_mapping: the target for the generated modules_mapping.json file.
         manifest: the target for the Gazelle manifest file.
     """
+    if pip_deps_repository_name != "":
+        # buildifier: disable=print
+        print("DEPRECATED pip_deps_repository_name in //{}:{}. Please use pip_repository_name instead.".format(
```

Comment:
do we need this deprecation here too?

Reply:
Yes, it communicates the users to take an action rather than ignore the deprecated property until the deprecation is gone and it becomes a breaking change with no warnings on how to fix it.